### PR TITLE
Gh 885 reduce times byte array is copied

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

**Comment by GCHQDeveloper404** [comment]
Can't reopen the old pull request, but will review comments.

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java [hunk 1]
```diff
@@ -61,57 +60,52 @@ public ByteEntityAccumuloElementConverter(final Schema schema) {
...
+        byte[] rowKey = new byte[carriage + secondLen + 5];
+        System.arraycopy(first, 0, rowKey, 0, carriage);
+        rowKey[carriage++] = ByteArrayEscapeUtils.DELIMITER;
+        rowKey[carriage++] = directionFlag;
+        rowKey[carriage++] = ByteArrayEscapeUtils.DELIMITER;
+        System.arraycopy(second, 0, rowKey, carriage, secondLen);
+        carriage += secondLen;
+        rowKey[carriage++] = ByteArrayEscapeUtils.DELIMITER;
+        rowKey[carriage] = directionFlag;
+        //carriage++;
```
Comment (p013570) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java]: Can you remove this carriage++ comment?

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java [hunk 2]
```diff
@@ -39,27 +39,34 @@ public boolean canHandle(final Class clazz) {
...
         final long value = Double.doubleToRawLongBits(d);
-        out[0] = (byte) ((int) (value & 255));
-        out[1] = (byte) ((int) (value >> 8) & 255);
-        out[2] = (byte) ((int) (value >> 16) & 255);
-        out[3] = (byte) ((int) (value >> 24) & 255);
-        out[4] = (byte) ((int) (value >> 32) & 255);
-        out[5] = (byte) ((int) (value >> 40) & 255);
-        out[6] = (byte) ((int) (value >> 48) & 255);
-        out[7] = (byte) ((int) (value >> 56) & 255);
+        out[0] = (byte) ((value & 255));
```
Comment (p013570) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java]: I'm not sure about these casting changes in these files. Can we leave it as it was?

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java (hunk 1 above)
Comment (GCHQDeveloper404) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java]: I had kept this comment in there as a visual clue that if the carriage is to be used again it needs appending.

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java (hunk 2 above)
Comment (GCHQDeveloper404) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java]: You have long, being cut down to a int being cut down to byte?
whats the difference with long just being cut down to a byte?

Comment (GCHQDeveloper404) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java]: @Test
    public void shouldByteTheSame() throws Exception {
        final long value = new Date().getTime();
        long value2 = new Long(value);

        final byte[] out = new byte[8];
        out[0] = (byte) ((int) (value >> 56) & 255);
        out[1] = (byte) ((int) (value >> 48) & 255);
        out[2] = (byte) ((int) (value >> 40) & 255);
        out[3] = (byte) ((int) (value >> 32) & 255);
        out[4] = (byte) ((int) (value >> 24) & 255);
        out[5] = (byte) ((int) (value >> 16) & 255);
        out[6] = (byte) ((int) (value >> 8) & 255);
        out[7] = (byte) ((int) value & 255);

        final byte[] out2 = new byte[8];
        out2[0] = (byte) ((value2 >> 56) & 255);
        out2[1] = (byte) ((value2 >> 48) & 255);
        out2[2] = (byte) ((value2 >> 40) & 255);
        out2[3] = (byte) ((value2 >> 32) & 255);
        out2[4] = (byte) ((value2 >> 24) & 255);
        out2[5] = (byte) ((value2 >> 16) & 255);
        out2[6] = (byte) ((value2 >> 8) & 255);
        out2[7] = (byte) (value2 & 255);

        assertEquals(value, value2);
        assertArrayEquals(out, out2);
    }

Comment (GCHQDeveloper404) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java]: No  :P

```
@Test
    public void shouldByteTheSame() throws Exception {
        final long value = new Date().getTime();
        long value2 = new Long(value);

        final byte[] out = new byte[8];
        out[0] = (byte) ((int) (value >> 56) & 255);
        out[1] = (byte) ((int) (value >> 48) & 255);
        out[2] = (byte) ((int) (value >> 40) & 255);
        out[3] = (byte) ((int) (value >> 32) & 255);
        out[4] = (byte) ((int) (value >> 24) & 255);
        out[5] = (byte) ((int) (value >> 16) & 255);
        out[6] = (byte) ((int) (value >> 8) & 255);
        out[7] = (byte) ((int) value & 255);

        final byte[] out2 = new byte[8];
        out2[0] = (byte) ((value2 >> 56) & 255);
        out2[1] = (byte) ((value2 >> 48) & 255);
        out2[2] = (byte) ((value2 >> 40) & 255);
        out2[3] = (byte) ((value2 >> 32) & 255);
        out2[4] = (byte) ((value2 >> 24) & 255);
        out2[5] = (byte) ((value2 >> 16) & 255);
        out2[6] = (byte) ((value2 >> 8) & 255);
        out2[7] = (byte) (value2 & 255);

        assertEquals(value, value2);
        assertArrayEquals(out, out2);
    }
```

**Comment by GCHQDeveloper404** [comment]
@gaffer01 This is latest branch I'd like you to take a look at.

File: core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java (hunk 2 above)
Comment (GCHQDeveloper404) [inline core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/implementation/raw/RawDoubleSerialiser.java]: Don't fear the metalic currency (change)

File: store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java (hunk 1 above)
Comment (p013570) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java]: Perhaps just put a comment explaining that instead.

Comment (GCHQDeveloper404) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java]: Comments?? that is so verbose!

Comment (GCHQDeveloper404) [inline store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java]: deleted.

**Pull request merged**